src/common.ts

```diff
@@ -190,35 +190,48 @@ export function isMessage(val: unknown): val is Message {
   return false;
 }
 
+export type JSONReviver = (
+  this: unknown,
+  key: string,
+  value: unknown,
+) => unknown;
+
 /**
  * Parses the raw websocket message data to a valid message.
  *
  * @category Common
  */
-export function parseMessage(data: unknown): Message {
+export function parseMessage(data: unknown, reviver?: JSONReviver): Message {
   if (isMessage(data)) {
     return data;
   }
   if (typeof data !== 'string') {
     throw new Error('Message not parsable');
   }
-  const message = JSON.parse(data);
+  const message = JSON.parse(data, reviver);
   if (!isMessage(message)) {
     throw new Error('Invalid message');
   }
   return message;
 }
 
+export type JSONReplacer = (
```

Comment:
I feel like `JSONMessageReplacer` is a better and more explicit name. You're altering the stringification process of exactly only transported messages. Also, favouring documentation, a description would be nice.

```suggestion

/**
 * Function that alters the behavior of the outgoing `Message`
 * stringification process.
 *
 * Read more about using it:
 * https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify#the_replacer_parameter
 *
 * @category Common
 */
export type JSONMessageReplacer = (
```